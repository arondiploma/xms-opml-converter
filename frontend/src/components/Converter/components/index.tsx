export { default as Uploader } from "./Uploader";
export { default as ResultViewer } from "./ResultViewer";