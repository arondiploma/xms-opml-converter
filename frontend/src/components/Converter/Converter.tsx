import React, { useState } from 'react';
import { Uploader, ResultViewer } from "./components";

const Converter: React.FC = () => {

    const [results, setResult] = useState<Array<any>>([]);
    const handleResult = (convertResult: Array<any>) => {
        setResult(convertResult);
    }
    const handleNew = () => {
        setResult([]);
    }

    return (
        <>
            {
                results.length === 0 ?
                    (<Uploader onConvertSuccess={handleResult} />) :
                    (<ResultViewer results={results} onNew={handleNew} />)
            }
        </>
    );
};

export default Converter;