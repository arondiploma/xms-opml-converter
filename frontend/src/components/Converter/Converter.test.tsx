import React from 'react';
import Converter from "./Converter";
import renderer from 'react-test-renderer';

describe("<Converter/>", () => {
    it("test", () => {
        const component = renderer.create(
            <Converter />,
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});