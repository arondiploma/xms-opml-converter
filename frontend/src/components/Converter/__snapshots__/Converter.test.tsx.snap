// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Converter/> test 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthMd"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-1 MuiPaper-rounded"
  >
    <div
      className="MuiCardContent-root"
    >
      <label
        htmlFor="input-file"
      >
        <div
          className="makeStyles-button-40 makeStyles-buttonNoFile-39"
        >
          <input
            className="makeStyles-hidden-38"
            disabled={false}
            id="input-file"
            multiple={true}
            onChange={[Function]}
            type="file"
          />
          <div>
            <img
              alt="Select files"
              className="makeStyles-file_img-41"
              src="/images/undraw_add_file2_gvbb.svg"
            />
          </div>
          <div>
            <h3
              className="MuiTypography-root MuiTypography-h3"
            >
              Select files
            </h3>
            <p
              className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
            >
              Click here to open file browser.
            </p>
          </div>
        </div>
      </label>
      <ul
        className="MuiList-root MuiList-padding"
      />
    </div>
    <div
      className="MuiCardActions-root makeStyles-controls-79 makeStyles-hidden-80 MuiCardActions-spacing"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Clear All
        </span>
      </button>
      <div
        className="makeStyles-wrapper-81"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Convert
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
